shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -274,6 +188,20 @@ impl ObjectFetcher {
             });
         }
 
+        if piece_offset >= RawRecord::SIZE as u32 {
+            tracing::debug!(
+                %piece_index,
+                piece_offset,
+                RawRecord_SIZE = RawRecord::SIZE,
+                "Invalid piece offset for object: must be less than the size of a raw record",
+            );
```

Comment:
I'd say we don't need these logs if we already return an error explicitly. It balloons the code side and brings little to no benefit.

Reply:
I agree the logs don’t need to be in there long-term, but I’d like to keep them temporarily while I’m testing and debugging this code.